/cc @kamilchm 

I've tried to adapt the patches to the new version, though there are still build errors left:

<details>
 <summary>Build Output</summary>

```
$ nix-build . -A qtile
these derivations will be built:
  /nix/store/xhwsqs95886livj1bx19d0cp6ss8zcd6-qtile-0.10.7.drv
building '/nix/store/xhwsqs95886livj1bx19d0cp6ss8zcd6-qtile-0.10.7.drv'...
unpacking sources
unpacking source archive /nix/store/iaw2gfsab16nba98q9wvv3w9789bb7yi-qtile
source root is qtile
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file qtile/tox.ini
patching sources
configuring
building
warning: no previously-included files found matching 'xcffib/_ffi.py'

Installed /tmp/nix-build-qtile-0.10.7.drv-0/qtile/.eggs/xcffib-0.5.1-py2.7.egg
Traceback (most recent call last):
  File "nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 170, in <module>
    **cffi_args
  File "/nix/store/a4zb290zkx8avpand4a58ndci8ppczik-python2.7-bootstrapped-pip-9.0.1/lib/python2.7/site-packages/setuptools/__init__.py", line 129, in setup
    return distutils.core.setup(**attrs)
  File "/nix/store/87l8l9w3x8saxqz9sfcfw2i3jiilmxfi-python-2.7.14/lib/python2.7/distutils/core.py", line 111, in setup
    _setup_distribution = dist = klass(attrs)
  File "/nix/store/a4zb290zkx8avpand4a58ndci8ppczik-python2.7-bootstrapped-pip-9.0.1/lib/python2.7/site-packages/setuptools/dist.py", line 333, in __init__
    _Distribution.__init__(self, attrs)
  File "/nix/store/87l8l9w3x8saxqz9sfcfw2i3jiilmxfi-python-2.7.14/lib/python2.7/distutils/dist.py", line 287, in __init__
    self.finalize_options()
  File "/nix/store/a4zb290zkx8avpand4a58ndci8ppczik-python2.7-bootstrapped-pip-9.0.1/lib/python2.7/site-packages/setuptools/dist.py", line 476, in finalize_options
    ep.load()(self, ep.name, value)
  File "/nix/store/ibgvw4nhlvc0kpd88b789qbynhnjym5z-python2.7-cffi-1.11.4/lib/python2.7/site-packages/cffi/setuptools_ext.py", line 193, in cffi_modules
    add_cffi_module(dist, cffi_module)
  File "/nix/store/ibgvw4nhlvc0kpd88b789qbynhnjym5z-python2.7-cffi-1.11.4/lib/python2.7/site-packages/cffi/setuptools_ext.py", line 49, in add_cffi_module
    execfile(build_file_name, mod_vars)
  File "/nix/store/ibgvw4nhlvc0kpd88b789qbynhnjym5z-python2.7-cffi-1.11.4/lib/python2.7/site-packages/cffi/setuptools_ext.py", line 25, in execfile
    exec(code, glob, glob)
  File "libqtile/ffi_build.py", line 25, in <module>
    from xcffib.ffi_build import ffi as xcffib_ffi
  File "/tmp/nix-build-qtile-0.10.7.drv-0/qtile/.eggs/xcffib-0.5.1-py2.7.egg/xcffib/__init__.py", line 34, in <module>
    lib = ffi.dlopen(soname)
OSError: cannot load library 'libxcb.so.1': libxcb.so.1: cannot open shared object file: No such file or directory
builder for '/nix/store/xhwsqs95886livj1bx19d0cp6ss8zcd6-qtile-0.10.7.drv' failed with exit code 1
error: build of '/nix/store/xhwsqs95886livj1bx19d0cp6ss8zcd6-qtile-0.10.7.drv' failed
```

</details>

--- a/pkgs/applications/window-managers/qtile/0001-Substitution-vars-for-absolute-paths.patch
+++ b/pkgs/applications/window-managers/qtile/0001-Substitution-vars-for-absolute-paths.patch
@@ -1,15 +1,5 @@
-From 00c5af939567429d40877845dc52b54fde2d8a50 Mon Sep 17 00:00:00 2001
-From: "Alexander V. Nikolaev" <avn@avnik.info>
-Date: Thu, 26 Nov 2015 10:53:12 +0200
-Subject: [PATCH 1/3] Substitution vars for absolute paths
-
----
- libqtile/pangocffi.py | 6 +++---
- libqtile/xcursors.py  | 2 +-
- 2 files changed, 4 insertions(+), 4 deletions(-)
-
 diff --git a/libqtile/pangocffi.py b/libqtile/pangocffi.py
-index 27691d1..25f690d 100644
+index 1e8f5c04..e860d43a 100644
 --- a/libqtile/pangocffi.py
 +++ b/libqtile/pangocffi.py
 @@ -58,9 +58,9 @@ except ImportError:
@@ -26,18 +16,15 @@ index 27691d1..25f690d 100644
  
  def CairoContext(cairo_t):
 diff --git a/libqtile/xcursors.py b/libqtile/xcursors.py
-index e0e55e1..59b6428 100644
+index f1133555..3e61204a 100644
 --- a/libqtile/xcursors.py
 +++ b/libqtile/xcursors.py
-@@ -114,7 +114,7 @@ class Cursors(dict):
+@@ -112,7 +112,7 @@ class Cursors(dict):
  
      def _setup_xcursor_binding(self):
          try:
 -            xcursor = ffi.dlopen('libxcb-cursor.so')
 +            xcursor = ffi.dlopen('@xcb-cursor@/lib/libxcb-cursor.so')
          except OSError:
-             self.log.warning("xcb-cursor not found, fallback to font pointer")
+             logger.warning("xcb-cursor not found, fallback to font pointer")
              return False
--- 
-2.6.3
-

--- a/pkgs/applications/window-managers/qtile/0002-Restore-PATH-and-PYTHONPATH.patch
+++ b/pkgs/applications/window-managers/qtile/0002-Restore-PATH-and-PYTHONPATH.patch
@@ -1,57 +1,52 @@
-From f299a0aa0eefcf16bb4990f00ac3946727f43ef3 Mon Sep 17 00:00:00 2001
-From: "Alexander V. Nikolaev" <avn@avnik.info>
-Date: Fri, 27 Nov 2015 10:49:48 +0200
-Subject: [PATCH 2/3] Restore PATH and PYTHONPATH
-
----
- bin/qtile         | 1 +
- bin/qtile-run     | 1 +
- bin/qtile-session | 2 ++
- libqtile/utils.py | 7 +++++++
- 4 files changed, 11 insertions(+)
-
+diff --git a/bin/qshell b/bin/qshell
+index 2ba7e61c..0ac2a2ef 100755
+--- a/bin/qshell
++++ b/bin/qshell
+@@ -28,5 +28,6 @@ base_dir = os.path.abspath(os.path.join(this_dir, ".."))
+ sys.path.insert(0, base_dir)
+ 
+ if __name__ == '__main__':
++    __import__("importlib").import_module("libqtile.utils").restore_os_environment()
+     from libqtile.scripts import qshell
+     qshell.main()
 diff --git a/bin/qtile b/bin/qtile
-index 66034fe..ce3fcd1 100755
+index 3e82814d..335b5cea 100755
 --- a/bin/qtile
 +++ b/bin/qtile
-@@ -131,6 +131,7 @@ def make_qtile():
+@@ -29,5 +29,6 @@ base_dir = os.path.abspath(os.path.join(this_dir, ".."))
+ sys.path.insert(0, base_dir)
  
- 
- if __name__ == "__main__":
+ if __name__ == '__main__':
 +    __import__("importlib").import_module("libqtile.utils").restore_os_environment()
-     rename_process()
-     q = make_qtile()
-     try:
+     from libqtile.scripts import qtile
+     qtile.main()
 diff --git a/bin/qtile-run b/bin/qtile-run
-index ccedb96..646a476 100755
+index e4b121be..1c203bc9 100755
 --- a/bin/qtile-run
 +++ b/bin/qtile-run
-@@ -50,6 +50,7 @@ def main():
-     proc.wait()
+@@ -8,5 +8,6 @@ base_dir = os.path.abspath(os.path.join(this_dir, ".."))
+ sys.path.insert(0, base_dir)
  
- if __name__ == "__main__":
+ if __name__ == '__main__':
 +    __import__("importlib").import_module("libqtile.utils").restore_os_environment()
-     try:
-         main()
-     except KeyboardInterrupt:
-diff --git a/bin/qtile-session b/bin/qtile-session
-index 84f6a2d..da31b12 100755
---- a/bin/qtile-session
-+++ b/bin/qtile-session
-@@ -25,6 +25,8 @@
- Qtile session manager.
- """
+     from libqtile.scripts import qtile_run
+     qtile_run.main()
+diff --git a/bin/qtile-top b/bin/qtile-top
+index 5316e0e7..272c6430 100755
+--- a/bin/qtile-top
++++ b/bin/qtile-top
+@@ -8,5 +8,6 @@ base_dir = os.path.abspath(os.path.join(this_dir, ".."))
+ sys.path.insert(0, base_dir)
  
-+__import__("importlib").import_module("libqtile.utils").restore_os_environment()
-+
- from libqtile.log_utils import init_log
- import logging
- import os
+ if __name__ == '__main__':
++    __import__("importlib").import_module("libqtile.utils").restore_os_environment()
+     from libqtile.scripts import qtile_top
+     qtile_top.main()
 diff --git a/libqtile/utils.py b/libqtile/utils.py
-index 284089b..ec3539e 100644
+index 36ed0a58..bca9eab3 100644
 --- a/libqtile/utils.py
 +++ b/libqtile/utils.py
-@@ -227,3 +227,11 @@ def describe_attributes(obj, attrs, func=None):
+@@ -240,3 +240,11 @@ def describe_attributes(obj, attrs, func=None):
              pairs.append('%s=%s' % (attr, value))
  
      return ', '.join(pairs)

--- a/pkgs/applications/window-managers/qtile/0003-Restart-executable.patch
+++ b/pkgs/applications/window-managers/qtile/0003-Restart-executable.patch
@@ -1,17 +1,8 @@
-From b560c11078fecc35df2c62f34beda06c4e80a10d Mon Sep 17 00:00:00 2001
-From: "Alexander V. Nikolaev" <avn@avnik.info>
-Date: Fri, 27 Nov 2015 10:54:35 +0200
-Subject: [PATCH 3/3] Restart executable
-
----
- libqtile/manager.py | 2 +-
- 1 file changed, 1 insertion(+), 1 deletion(-)
-
 diff --git a/libqtile/manager.py b/libqtile/manager.py
-index b1a38e2..110f7d8 100644
+index 36518a74..9b6bdd02 100644
 --- a/libqtile/manager.py
 +++ b/libqtile/manager.py
-@@ -1339,7 +1339,7 @@ class Qtile(command.CommandObject):
+@@ -1386,7 +1386,7 @@ class Qtile(command.CommandObject):
          argv = [s for s in argv if not s.startswith('--with-state')]
          argv.append('--with-state=' + buf.getvalue().decode())
  
@@ -19,7 +10,4 @@ index b1a38e2..110f7d8 100644
 +        self.cmd_execute(os.environ.get("QTILE_WRAPPER", "@out@/bin/qtile"), argv[1:])
  
      def cmd_spawn(self, cmd):
-         """
--- 
-2.6.3
-
+         """Run cmd in a shell.

--- a/pkgs/applications/window-managers/qtile/default.nix
+++ b/pkgs/applications/window-managers/qtile/default.nix
@@ -7,13 +7,13 @@ in
 
 python27Packages.buildPythonApplication rec {
   name = "qtile-${version}";
-  version = "0.10.4";
+  version = "0.10.7";
 
   src = fetchFromGitHub {
     owner = "qtile";
     repo = "qtile";
     rev = "v${version}";
-    sha256 = "0rwklzgkp3x242xql6qmfpfnhr788hd3jc1l80pc5ybxlwyfx59i";
+    sha256 = "18szgplyym0b65vnaa8nqzadq6q0mhsiky9g5hqhn7xzf4kykmj8";
   };
 
   patches = [

--- a/pkgs/development/python-modules/xcffib/default.nix
+++ b/pkgs/development/python-modules/xcffib/default.nix
@@ -1,0 +1,31 @@
+{ stdenv
+, buildPythonPackage
+, fetchPypi
+, xorg
+, cffi
+, six
+}:
+
+buildPythonPackage rec {
+  version = "0.5.1";
+  pname = "xcffib";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "09gbnmr5vn58mm8xi3fmd7fz6743cks6c46dphnxzwax6zsxmy60";
+  };
+
+  patchPhase = ''
+    # Hardcode cairo library path
+    sed -e 's,ffi\.dlopen(,&"${xorg.libxcb.out}/lib/" + ,' -i xcffib/__init__.py
+  '';
+
+  propagatedBuildInputs = [ cffi six ];
+
+  meta = with stdenv.lib; {
+    description = "A drop in replacement for xpyb, an XCB python binding";
+    homepage = "https://github.com/tych0/xcffib";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ kamilchm ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -21012,29 +21012,7 @@ EOF
 
   pluggy = callPackage ../development/python-modules/pluggy {};
 
-  xcffib = buildPythonPackage rec {
-    version = "0.3.2";
-    name = "xcffib-${version}";
-
-    src = pkgs.fetchurl {
-      url = "mirror://pypi/x/xcffib/${name}.tar.gz";
-      sha256 = "a84eecd5a1bb7570e26c83aca87a2016578ca4e353e1fa56189e95bdef063e6a";
-    };
-
-    patchPhase = ''
-      # Hardcode cairo library path
-      sed -e 's,ffi\.dlopen(,&"${pkgs.xorg.libxcb.out}/lib/" + ,' -i xcffib/__init__.py
-    '';
-
-    propagatedBuildInputs = [ self.cffi self.six ];
-
-    meta = {
-      description = "A drop in replacement for xpyb, an XCB python binding";
-      homepage = "https://github.com/tych0/xcffib";
-      license = licenses.asl20;
-      maintainers = with maintainers; [ kamilchm ];
-    };
-  };
+  xcffib = callPackage ../development/python-modules/xcffib {};
 
   pafy = callPackage ../development/python-modules/pafy { };
 

